packages/calcite-components/src/components/table/table.e2e.ts

```diff
@@ -2739,4 +2740,52 @@ describe("keyboard navigation", () => {
       await page.$eval(`#${row3.id}`, (el) => el.shadowRoot?.activeElement.shadowRoot?.querySelector("td").classList),
     ).toEqual({ "0": CSS.selectionCell });
   });
+
+  it("updates table-row's selected property correctly when calciteTabRowSelect event is emitted", async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      html`<calcite-table caption="Simple table" selection-mode="multiple">
+        <calcite-table-row slot="table-header">
+          <calcite-table-header heading="Heading" description="Description"></calcite-table-header>
+          <calcite-table-header heading="Heading" description="Description"></calcite-table-header>
+        </calcite-table-row>
+        <calcite-table-row id="row-1">
+          <calcite-table-cell>row1</calcite-table-cell>
+          <calcite-table-cell>row1</calcite-table-cell>
+        </calcite-table-row>
+        <calcite-table-row id="row-2">
+          <calcite-table-cell>row2</calcite-table-cell>
+          <calcite-table-cell>row2</calcite-table-cell>
+        </calcite-table-row>
+      </calcite-table>`,
+    );
+
+    const row1 = await page.find("calcite-table-row");
+    expect(await row1.getProperty("selected")).toBe(false);
+    type TestWindow = typeof window & { selectedValue: boolean };
+
+    await page.evaluate(() => {
+      document.querySelector("calcite-table").addEventListener("calciteTableRowSelect", (event) => {
+        (window as TestWindow).selectedValue = (event.target as TableRow["el"]).selected;
+      });
+    });
+
+    await page.$eval("calcite-table", () => {
+      const row = document.getElementById("row-1");
+      const cell = row.shadowRoot.querySelector<TableCell["el"]>("calcite-table-cell:first-child");
+      cell.click();
+    });
+
+    await page.waitForChanges();
```

Comment:
Nitpick: can you move the `waitForChanges` calls to be right after eval? These usually follow a browser-context update.